packages/hop-node/src/vault/Vault.ts

```diff
@@ -0,0 +1,33 @@
+import { BigNumber } from 'ethers'
+import { Chain } from 'src/constants'
+import { YearnVault } from './YearnVault'
+
+export abstract class Vault {
+  static from (chain: Chain, token: string, signer: any) {
+    const yearnVaultTokens = new Set(['ETH', 'USDC', 'USDT', 'DAI'])
+    const useYearnVault = chain === Chain.Ethereum && yearnVaultTokens.has(token)
+    if (useYearnVault) {
+      return new YearnVault(chain, token, signer)
+    }
+  }
+
+  async getBalance (account?: string): Promise<BigNumber> {
```

Comment:
Super minor but if you mark these functions as abstract instead of throwing an error, any missing function definitions in child classes will be caught in compile time rather than run time

Reply:
oh nice, works perfectly

Resolved with https://github.com/hop-protocol/hop/pull/404/commits/19508919855581f7770c150704123ebefffb6c65